examples/thompson_sampling.py

```diff
@@ -0,0 +1,302 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: Gaussian Process
+=========================
+
+In this example we show how to implement Thompson sampling for Gaussian processes.
+
+"""
+
+import argparse
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import jax
+import jax.numpy as jnp
+import jax.random as random
+
+import numpyro
+import numpyro.distributions as dist
+from numpyro.infer import SVI, Trace_ELBO
+from numpyro.infer.autoguide import AutoDelta
+
+numpyro.enable_x64()
+
+
+# defining the function. At y=0 to get a 1D cut at the origin
+def ackley_1d(x, y=0):
+    out = (
+        -20 * jnp.exp(-0.2 * jnp.sqrt(0.5 * (x ** 2 + y ** 2)))
+        - jnp.exp(0.5 * (jnp.cos(2 * jnp.pi * x) + jnp.cos(2 * jnp.pi * y)))
+        + jnp.e
+        + 20
+    )
+    return out
+
+
+# matern kernel with nu = 1.5
```

Comment:
this does not appear to be the 3/2 kernel?

Reply:
You're right, it's the 5/2 kernel. I fixed the comment.